src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -1153,7 +1152,7 @@ event::ptr primitive_inst::update_weights() {
             GPU_DEBUG_PROFILED_STAGE_CACHE_HIT(true);
             GPU_DEBUG_TRACE_DETAIL << id() << ": reinterpret original weights memory from " << original_layout.to_short_string()
                                            << " to " << expected_layout.to_short_string() << std::endl;
-            _reordered_weights_cache.add(expected_layout, engine.reinterpret_buffer(*original_weights_memory, expected_layout));
+            _impl_params->weights_layout = optional_layout(original_layout);
```

Comment:
Why is the `weights_layout` set to `original_layout` instead of `expected_layout`?

Reply:
Ah, they are physically same (compatible) so no matter which is set. 
However I will set it as expected layout to be more explicitly. Thanks!

Reply:
Oh, if so, this line can be deleted, because it's already set by the above Line 1145.

Reply:
Applied thanks!